Null values published for 'apiName' and 'destination' attrs

**Description:**
Below error is logged in wso2am-analytics-2.6.0 when processing MGW published stats.
```
osgi> [2019-05-23 16:07:37,782]  INFO {org.wso2.analytics.apim.rest.api.file.impl.UsageApiServiceImpl} - Successfully uploaded the API Usage file [api-usage-data.dat.1558607852312.bffb34ac-e3bc-4da8-b2af-4b3723c49360.zip]
[2019-05-23 16:08:21,726]  INFO {org.wso2.extension.siddhi.io.mgwfile.MGWFileDataRetriever} - Started publishing API usage in file : [ FileName : api-usage-data.dat.1558607852312.bffb34ac-e3bc-4da8-b2af-4b3723c49360.zip, TimeStamp : 1558607852000]
[2019-05-23 16:08:21,789]  INFO {org.wso2.extension.siddhi.io.mgwfile.MGWFileDataRetriever} - Completed publishing API Usage from file : [ FileName : api-usage-data.dat.1558607852312.bffb34ac-e3bc-4da8-b2af-4b3723c49360.zip, TimeStamp : 1558607852000]
[2019-05-23 16:08:22,007] ERROR {org.wso2.siddhi.core.aggregation.IncrementalExecutor} - Exception occurred when writing to aggregation table of duration 'SECONDS'. This should be investigated as this can cause accuracy loss. org.wso2.extension.siddhi.store.rdbms.exception.RDBMSTableException: Error in adding events to 'ApiPerDestinationAgg_SECONDS' store: Column 'destination' cannot be null
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.add(RDBMSEventTable.java:562)
	at org.wso2.siddhi.core.table.record.AbstractRecordTable.add(AbstractRecordTable.java:98)
	at org.wso2.siddhi.core.table.Table.addEvents(Table.java:139)
	at org.wso2.siddhi.core.aggregation.IncrementalExecutor.lambda$dispatchEvents$2(IncrementalExecutor.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
...
```
<details>
<summary>Complete log...</summary>

```
Caused by: java.sql.BatchUpdateException: Column 'destination' cannot be null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.Util.getInstance(Util.java:408)
	at com.mysql.jdbc.SQLError.createBatchUpdateException(SQLError.java:1163)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1823)
	at com.mysql.jdbc.PreparedStatement.executeBatchInternal(PreparedStatement.java:1307)
	at com.mysql.jdbc.StatementImpl.executeBatch(StatementImpl.java:970)
	at com.zaxxer.hikari.proxy.StatementProxy.executeBatch(StatementProxy.java:127)
	at com.zaxxer.hikari.proxy.HikariPreparedStatementProxy.executeBatch(HikariPreparedStatementProxy.java)
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.batchExecuteQueriesWithRecords(RDBMSEventTable.java:1418)
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.add(RDBMSEventTable.java:559)
	... 6 more
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Column 'destination' cannot be null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.Util.getInstance(Util.java:408)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2495)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1903)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2124)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1801)
	... 12 more

[2019-05-23 16:08:22,008] ERROR {org.wso2.siddhi.core.aggregation.IncrementalExecutor} - Exception occurred when writing to aggregation table of duration 'SECONDS'. This should be investigated as this can cause accuracy loss. org.wso2.extension.siddhi.store.rdbms.exception.RDBMSTableException: Error in adding events to 'APIM_ReqCountAgg_SECONDS' store: Column 'apiName' cannot be null
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.add(RDBMSEventTable.java:562)
	at org.wso2.siddhi.core.table.record.AbstractRecordTable.add(AbstractRecordTable.java:98)
	at org.wso2.siddhi.core.table.Table.addEvents(Table.java:139)
	at org.wso2.siddhi.core.aggregation.IncrementalExecutor.lambda$dispatchEvents$2(IncrementalExecutor.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.sql.BatchUpdateException: Column 'apiName' cannot be null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.Util.getInstance(Util.java:408)
	at com.mysql.jdbc.SQLError.createBatchUpdateException(SQLError.java:1163)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1823)
	at com.mysql.jdbc.PreparedStatement.executeBatchInternal(PreparedStatement.java:1307)
	at com.mysql.jdbc.StatementImpl.executeBatch(StatementImpl.java:970)
	at com.zaxxer.hikari.proxy.StatementProxy.executeBatch(StatementProxy.java:127)
	at com.zaxxer.hikari.proxy.HikariPreparedStatementProxy.executeBatch(HikariPreparedStatementProxy.java)
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.batchExecuteQueriesWithRecords(RDBMSEventTable.java:1418)
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.add(RDBMSEventTable.java:559)
	... 6 more
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Column 'apiName' cannot be null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.Util.getInstance(Util.java:408)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2495)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1903)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2124)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1801)
	... 12 more
```
</details>

**OS, DB, other environment details and versions:**    
WSO2 APIM 2.6.0 with APIM Analytics 2.6.0

**Steps to reproduce:**
- Configure WSO2 APIM analytics 2.6.0 with MGW
- Build a MGW project
- Send some requests

--- a/components/micro-gateway-cli/src/main/resources/templates/service.mustache
+++ b/components/micro-gateway-cli/src/main/resources/templates/service.mustache
@@ -2,11 +2,8 @@ import ballerina/log;
 import ballerina/http;
 import ballerina/config;
 import ballerina/time;
-
 {{#if containerConfig.hasKubernetes}}import ballerinax/kubernetes;{{/if}}
-
 import wso2/gateway;
-
 {{#equals endpointConfig.prodEndpoints.type "http"}}
     {{#endpointConfig.prodEndpoints}}{{>httpEndpoint}}{{/endpointConfig.prodEndpoints}}
 {{/equals}}
@@ -133,12 +130,12 @@ service {{cut qualifiedServiceName " "}} on {{#api.transport}}{{#equals . "http"
                 {{else}}
                     {{>noSandAvailableError}}
                 {{/if}}
-                    }
+                }
             {{/endpointConfig.sandEndpoints}}
         {{/endpointConfig.prodEndpoints}}
         {{^endpointConfig.prodEndpoints}}
             {{#endpointConfig.sandEndpoints}}
-                    if("SANDBOX" == <string>runtime:getInvocationContext().attributes["KEY_TYPE"]) {
+                if("SANDBOX" == <string>runtime:getInvocationContext().attributes["KEY_TYPE"]) {
                 {{#if hasSandEpConfig}}
                     {{>useResourceSandEp}}
                 {{else}}
@@ -150,21 +147,21 @@ service {{cut qualifiedServiceName " "}} on {{#api.transport}}{{#equals . "http"
                 {{else}}
                             {{>noProdAvailableError}}
                 {{/if}}
-                    }
+                }
             {{/endpointConfig.sandEndpoints}}
             {{^endpointConfig.sandEndpoints}}
                 {{#if hasSandEpConfig}}
-                        if("SANDBOX" == <string>runtime:getInvocationContext().attributes["KEY_TYPE"]) {
+                    if("SANDBOX" == <string>runtime:getInvocationContext().attributes["KEY_TYPE"]) {
                     {{>useResourceSandEp}}
                         } else {
                     {{#if hasProdEpConfig}}
                     {{else}}
                         {{>noProdAvailableError}}
-                        }
+                    }
                     {{/if}}
                 {{else}}
                     {{#if hasProdEpConfig}}
-                            if("PRODUCTION" == <string>runtime:getInvocationContext().attributes["KEY_TYPE"]) {
+                    if("PRODUCTION" == <string>runtime:getInvocationContext().attributes["KEY_TYPE"]) {
                     {{/if}}
                 {{/if}}
                 {{#if hasProdEpConfig}}
@@ -182,16 +179,13 @@ service {{cut qualifiedServiceName " "}} on {{#api.transport}}{{#equals . "http"
 
 
         if(clientResponse is http:Response) {
-            runtime:getInvocationContext().attributes["destination"] = clientResponse.resolvedRequestedURI;
             {{#if responseInterceptor}}{{responseInterceptor}} (outboundEp, clientResponse);{{/if}}
             {{#if apiResponseInterceptor}}{{apiResponseInterceptor}} (outboundEp, clientResponse);{{/if}}
             var outboundResult = outboundEp->respond(clientResponse);
             if (outboundResult is error) {
                 log:printError("Error when sending response", err = outboundResult);
             }
-        }
-
-        else {
+        } else {
             http:Response res = new;
             res.statusCode = 500;
             string errorMessage = clientResponse.reason();

--- a/components/micro-gateway-cli/src/main/resources/templates/useAPIProdEp.mustache
+++ b/components/micro-gateway-cli/src/main/resources/templates/useAPIProdEp.mustache
@@ -6,3 +6,4 @@
 {{else}}
     clientResponse = {{qualifiedServiceName}}_prod->forward(urlPostfix, req);
 {{/if}}
+runtime:getInvocationContext().attributes["destination"] = "{{endpointConfig.prodEndpoints.endpoints.0.endpointUrl}}";

--- a/components/micro-gateway-cli/src/main/resources/templates/useApiSandEp.mustache
+++ b/components/micro-gateway-cli/src/main/resources/templates/useApiSandEp.mustache
@@ -6,3 +6,4 @@
 {{else}}
     clientResponse = {{qualifiedServiceName}}_sand->forward(urlPostfix, req);
 {{/if}}
+runtime:getInvocationContext().attributes["destination"] = "{{endpointConfig.sandEndpoints.endpoints.0.endpointUrl}}";

--- a/components/micro-gateway-cli/src/main/resources/templates/useResourceProdEp.mustache
+++ b/components/micro-gateway-cli/src/main/resources/templates/useResourceProdEp.mustache
@@ -6,3 +6,4 @@
 {{else}}
     clientResponse = {{operationId}}_prod->forward(urlPostfix, req);
 {{/if}}
+runtime:getInvocationContext().attributes["destination"] = "{{epConfig.prodEndpoints.endpoints.0.endpointUrl}}";

--- a/components/micro-gateway-cli/src/main/resources/templates/useResourceSandEp.mustache
+++ b/components/micro-gateway-cli/src/main/resources/templates/useResourceSandEp.mustache
@@ -6,3 +6,4 @@
 {{else}}
     clientResponse = {{operationId}}_sand->forward(urlPostfix, req);
 {{/if}}
+runtime:getInvocationContext().attributes["destination"] = "{{epConfig.sandEndpoints.endpoints.0.endpointUrl}}";

--- a/components/micro-gateway-core/src/main/ballerina/gateway/responsetime_util.bal
+++ b/components/micro-gateway-core/src/main/ballerina/gateway/responsetime_util.bal
@@ -76,6 +76,7 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
 {
     RequestResponseExecutionDTO requestResponseExecutionDTO = {};
     boolean isSecured = <boolean>context.attributes[IS_SECURED];
+
     if (isSecured && context.attributes.hasKey(AUTHENTICATION_CONTEXT)) {
         AuthenticationContext authContext = <AuthenticationContext>context.attributes[AUTHENTICATION_CONTEXT];
         requestResponseExecutionDTO.apiCreator = authContext.apiPublisher;
@@ -100,7 +101,11 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
     // apim analytics requires context to be '<basePath>/<version>'
     string mgContext = getContext(context);
     mgContext = mgContext.split("/(?=$)")[0]; //split from last '/'
-    string analyticsContext = mgContext + "/" + requestResponseExecutionDTO.apiVersion;
+    string analyticsContext  = mgContext;
+
+    if (!mgContext.hasSuffix(requestResponseExecutionDTO.apiVersion)) {
+        analyticsContext = mgContext + "/" + requestResponseExecutionDTO.apiVersion;
+    }
 
     requestResponseExecutionDTO.apiContext = analyticsContext;
     requestResponseExecutionDTO.correlationId = <string>context.attributes[MESSAGE_ID];

--- a/components/micro-gateway-core/src/main/ballerina/gateway/responsetime_util.bal
+++ b/components/micro-gateway-core/src/main/ballerina/gateway/responsetime_util.bal
@@ -96,7 +96,13 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
     }
     requestResponseExecutionDTO.apiName = getApiName(context);
     requestResponseExecutionDTO.apiVersion = <string>apiConfigAnnotationMap[getServiceName(context.serviceName)].apiVersion;
-    requestResponseExecutionDTO.apiContext = getContext(context);
+
+    // apim analytics requires context to be '<basePath>/<version>'
+    string mgContext = getContext(context);
+    mgContext = mgContext.split("/(?=$)")[0]; //split from last '/'
+    string analyticsContext = mgContext + "/" + requestResponseExecutionDTO.apiVersion;
+
+    requestResponseExecutionDTO.apiContext = analyticsContext;
     requestResponseExecutionDTO.correlationId = <string>context.attributes[MESSAGE_ID];
 
     var res = response.cacheControl.noCache;

--- a/components/micro-gateway-core/src/main/ballerina/gateway/utils/utils.bal
+++ b/components/micro-gateway-core/src/main/ballerina/gateway/utils/utils.bal
@@ -312,7 +312,7 @@ public function getApiName(http:FilterContext context) returns (string) {
     if (apiName.contains("_")) {
         qualifiedName = apiName.replaceAll("_", "-");
     }
-    return qualifiedName;
+    return apiName;
 }
 
 public function getConfigValue(string instanceId, string property, string defaultValue) returns string {

--- a/components/micro-gateway-core/src/main/ballerina/gateway/utils/utils.bal
+++ b/components/micro-gateway-core/src/main/ballerina/gateway/utils/utils.bal
@@ -294,6 +294,7 @@ public function extractAccessToken(http:Request req, string authHeaderName) retu
 public function handleError(string message) returns (error) {
     return error(message);
 }
+
 public function getTenantDomain(http:FilterContext context) returns (string) {
     // todo: need to implement to get tenantDomain
     string apiContext = getContext(context);
@@ -305,13 +306,15 @@ public function getTenantDomain(http:FilterContext context) returns (string) {
         return SUPER_TENANT_DOMAIN_NAME;
     }
 }
+
 public function getApiName(http:FilterContext context) returns (string) {
-    string qualifiedName = "";
     string serviceName = context.serviceName;
     string apiName = serviceName.split("__")[0];
+
     if (apiName.contains("_")) {
-        qualifiedName = apiName.replaceAll("_", "-");
+        apiName = apiName.replaceAll("_", "-");
     }
+
     return apiName;
 }
 

